apps/meteor/tests/end-to-end/api/livechat/04-dashboards.ts

```diff
@@ -777,6 +778,198 @@ describe('LIVECHAT - dashboards', function () {
 		});
 	});
 
+	describe('[livechat/analytics/agent-overview] - Average first response time', () => {
+		let agent: { credentials: Credentials; user: IUser & { username: string } };
+		let originalFirstResponseTimeInSeconds: number;
+		let roomId: string;
+		const firstDelayInSeconds = 4;
+		const secondDelayInSeconds = 8;
+
+		before(async () => {
+			agent = await createAnOnlineAgent();
+		});
+
+		after(async () => {
+			await deleteUser(agent.user);
+		});
+
+		it('should return no average response time for an agent if no response has been sent in the period', async () => {
+			await startANewLivechatRoomAndTakeIt({ agent: agent.credentials });
+
+			const today = moment().startOf('day').format('YYYY-MM-DD');
+
+			const result = await request
+				.get(api('livechat/analytics/agent-overview'))
+				.query({ from: today, to: today, name: 'Avg_first_response_time' })
+				.set(credentials)
+				.expect('Content-Type', 'application/json')
+				.expect(200);
+
+			expect(result.body).to.have.property('success', true);
+			expect(result.body).to.have.property('head');
+			expect(result.body).to.have.property('data');
+			expect(result.body.data).to.be.an('array');
+			expect(result.body.data).to.not.deep.include({ name: agent.user.username });
```

Comment:
if the test returns empty data, this check will return true all the time. Can you assert that the array has to have at least 1 element?

Reply:
hmmm I'm thinking, is it really safe to assert that? At this point in my tests, I'm not even sure other agents exists (so that they can be returned here) -- so if I add this check here, it will depend on the result/what's done on previous tests
wdyt? I'm thinking of validating this in the **next** test instead, where I'm sure `data` shouldn't be empty

Reply:
I added this validation to the next test case. Please tell me what you think, @KevLehman